### PR TITLE
Added Pagination

### PR DESCRIPTION
**Pull request by Gaurav-git06**

### CODE REVIEW

**Review by ankitmashu — CHANGES_REQUESTED** [review]
1.  Check PMD and Checkstyle
2.  Docs not updated

File: src/main/java/iudx/catalogue/server/apiserver/MlayerApis.java [hunk 1]
```diff
@@ -334,25 +333,160 @@ public void createMlayerDomainHandler(RoutingContext routingContext) {
...
+//      requestParams.put(LIMIT,limitInt)
+//              .put(OFFSET,offsetInt)
+//              .put(ID,id);
+//      LOGGER.debug("394-> REQUEST PARAMS"+requestParams);
+//      HttpServerResponse response = routingContext.response();
+//      response.putHeader(HEADER_CONTENT_TYPE, MIME_APPLICATION_JSON);
+//      mlayerService.getMlayerDomain(
+//              requestParams,
+//              handler -> {
+//                  if (handler.succeeded()) {
```
Comment (ananjaykumar2) [inline src/main/java/iudx/catalogue/server/apiserver/MlayerApis.java:391]: i guess unnecessary comments/lines should be removed.

File: src/main/java/iudx/catalogue/server/apiserver/MlayerApis.java [hunk 2]
```diff
@@ -334,25 +333,160 @@ public void createMlayerDomainHandler(RoutingContext routingContext) {
...
+                  }
+              });  }
+
+    private JsonObject parseRequestParams(RoutingContext routingContext) {
+        LOGGER.debug("Info: parseRequestParams() started");
+
+        JsonObject requestParams = new JsonObject();
+        String id = routingContext.request().getParam("id");
+        String limit = routingContext.request().getParam("limit");
+        String offset = routingContext.request().getParam("offset");
```
Comment (ananjaykumar2) [inline src/main/java/iudx/catalogue/server/apiserver/MlayerApis.java:422]: it will be good to use constants instead of String.

**Review by ananjaykumar2 — CHANGES_REQUESTED** [review]

File: docs/openapi.yaml [hunk 3]
```diff
@@ -1693,13 +1693,31 @@ paths:
             `id` of the instance to be fetched
           schema:
             type: string
+        - name: limit
```
Comment (ankitmashu) [inline docs/openapi.yaml:1696]: Docs is not properly updated, you have to mention in introduction or in some other header that this bla bla api pagination introduced with maximum value and by default value to be allowed. Check other server docs

File: docs/openapi.yaml [hunk 4]
```diff
@@ -24,6 +24,18 @@ info:
...
       - The `<payload>` in the request sample could be replaced with the payload given adjacent to the request sample
       - The payload contains `<some_item_id>`, `<provider_id>`, `<some_item_name>`, for which the appropriate values could be added 
+    
+    # Pagination
+    Catalogue server also offers way to paginate the result for queries. If a query returns large number of records then user can use additional parameters in query parameters to limit numbers of records to be returned.
+    
+    - <b> Pagination is applicable only for</b>
+       - <b>/internal/ui/providers</b>
+       - <b>/internal/ui/instance</b>
+       - <b>/internal/ui/domain</b>
```
Comment (pranavrd) [inline docs/openapi.yaml:34]: Include `/search` API in this list

**Review by pranavrd — APPROVED** [review]

**Comment by pranavrd** [comment]
ok to test

**Comment by pranavrd** [comment]
retest this please

**Pull request merged**